src/osvr/Server/ConfigFilePaths.cpp

```diff
@@ -0,0 +1,79 @@
+// Internal Includes
+#include <osvr/Util/PlatformConfig.h>
+#include <osvr/Server/ConfigFilePaths.h>
+#include <osvr/Server/ConfigureServerFromFile.h>
+#include <osvr/Util/GetEnvironmentVariable.h>
+
+// Library/third-party includes
+#include <boost/filesystem.hpp>
+
+// Standard includes
+#include <vector>
+
+namespace osvr {
+    namespace server {
+
+        inline std::string getUserConfigDirectory() {
+            namespace fs = boost::filesystem;
+            using osvr::util::getEnvironmentVariable;
+
+            fs::path configDir;
+            std::string configSubpath = "config";
+
+#if defined(OSVR_LINUX)
+            // There's currently no great location for storing log files in the
+            // XDG system. (See the STATE proposal by Debian
+            // <https://wiki.debian.org/XDGBaseDirectorySpecification#Proposal:_STATE_directory>.)
+            // So for now, we'll store our log files in the $XDG_CACHE_HOME
+            // directory.
+            //
+            // $XDG_CACHE_HOME defines the base directory relative to which user
+            // specific non-essential data files should be stored. If
+            // $XDG_CACHE_HOME is either not set or empty, a default equal to
+            // $HOME/.cache should be used.
+            auto xdg_cache_dir = getEnvironmentVariable("XDG_CONFIG_HOME");
```

Comment:
On Linux, if you're searching for an existing configuration file (as opposed to determining where to write a new one), then you will want to check the `XDG_CONFIG_DIRS` paths as well. From the [XDG Base Directory Specification](https://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html):

> `$XDG_CONFIG_DIRS` defines the preference-ordered set of base directories to search for configuration files in addition to the `$XDG_CONFIG_HOME` base directory. The directories in `$XDG_CONFIG_DIRS` should be separated with a colon `:`.
>
> If `$XDG_CONFIG_DIRS` is either not set or empty, a value equal to `/etc/xdg` should be used.
>
> The order of base directories denotes their importance; the first directory listed is the most important. When the same information is defined in multiple places the information defined relative to the more important base directory takes precedent. [...] The base directory defined by `$XDG_CONFIG_HOME` is considered more important than any of the base directories defined by `$XDG_CONFIG_DIRS`.